js/utils/salt.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Generate 16 bytes salt for bcrypt by seed
+ *
+ * Should return the same salt for the same seed
+ *
+ * @param  {string}   seed The seed for salt
+ * @return {string}        The resulted salt
+ */
```

Comment:
As I understand salt on client can't be random and must be stable for username/password pair. This function generate stable compatible bcrypt salt. But I think my implementation is very naive and generated not "strong" salt, but I don't know how to do this better.